srcpkgs/systemd-boot/files/kernel.d/systemd-boot.post-install

```diff
@@ -0,0 +1,61 @@
+#!/bin/sh
+#
+# Kernel hook for systemd-boot.
+#
+# Arguments passed to this script: $1 pkgname, $2 version.
+#
+PKGNAME="$1"
+VERSION="$2"
+
+. "$ROOTDIR/etc/default/systemd-boot"
+
+if [ "$SYSTEMD_BOOT_DISABLE" ]; then
+	exit 0
+fi
+
+is_installed=$(bootctl is-installed)
+
+if [ "$is_installed" != "yes" ]; then
+	echo "systemd-boot is not installed to the ESP"
+	echo "run 'bootctl install' to install it"
+	exit 0
+fi
+
+#get major kernel ver, eg: linux6.8 -> 6.8
+major_ver=$(echo $PKGNAME | cut -c 6-)
+boot="$ROOTDIR/boot"
+esp="$ROOTDIR/boot/efi"
+entries="$esp/loader/entries"
+name="void-$major_ver"
+entry="$entries/$name.conf"
+options="$esp/loader/void-options.conf"
+loader="$esp/loader/loader.conf"
+uuid=$(cat /etc/fstab | awk '$2 == "/" { print $1 }' | sed 's/^UUID=//')
+
+[ -d "$esp" ] || exit 0
+
+mkdir -p "$entries"
+
+cat <<-EOF > "$entry"
+	title Void Linux $major_ver
+	version $VERSION
+	linux vmlinuz-void-$major_ver
+	initrd initramfs-void-$major_ver.img
+	options root=UUID=$uuid rw
```

Comment:
Is it possible not to hardcode `UUID`? Useful if someone uses `PARTUUID` for example.
See `gummiboot` hook for reference